Model/Sync/Reset.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace Yotpo\Core\Model\Sync;
+
+use Yotpo\Core\Model\Sync\Reset\Catalog;
+use Yotpo\Core\Model\Sync\Reset\Customers;
+use Yotpo\Core\Model\Sync\Reset\Orders;
+
+class Reset
+{
+    /**
+     * @var Catalog
+     */
+    protected $catalogReset;
+
+    /**
+     * @var Customers
+     */
+    protected $customersReset;
+
+    /**
+     * @var Orders
+     */
+    protected $ordersReset;
+
+    /**
+     * @param Catalog $catalogReset
+     * @param Customers $customersReset
+     * @param Orders $ordersReset
+     */
+    public function __construct(
+        Catalog $catalogReset,
+        Customers $customersReset,
+        Orders $ordersReset
+    ) {
+        $this->catalogReset = $catalogReset;
+        $this->customersReset = $customersReset;
+        $this->ordersReset = $ordersReset;
+    }
+
+    /**
+     * @param int $storeId
+     * @return void
+     */
+    public function resetSync($storeId)
+    {
+        $this->catalogReset->resetSync($storeId);
+        if (method_exists($this->customersReset, 'resetSync')) {
+            $this->customersReset->resetSync($storeId);
+        }
+        $this->ordersReset->resetSync($storeId);
```

Comment:
Please avoid code duplication
```suggestion
        $this->resetOrdersSync($storeId);
```

Reply:
This is done